_overviews/tutorials/scala-on-android.md

```diff
@@ -0,0 +1,137 @@
+
+partof: scala-on-android
+
+permalink: /tutorials/:title.html
+---
+
+By Maciej Gorywoda
+
+## Introduction
+The Android platform runs on Android Runtime which is a virtual machine based on JVM and, although not identical,[it's very similar to it](https://www.baeldung.com/java-jvm-vs-dvm). It's no surprise then that two of the most popular programming languages used to create Android apps are Java and Kotlin, both JVM languages. It might be a bit surprising though that Scala, also a JVM language and one in popularity [on par or ahead of Kotlin](https://redmonk.com/sogrady/2021/03/01/language-rankings-1-21/), has little presence on the Android platform. But it is possible to write Android apps in Scala, and in fact it's possible to do it in more than one way. Here, in this document, we will focus on how to write a modern Android app with Scala that uses GraalVM Native Image and JavaFX. At the end of this tutorial you will links to materials discussing other options. 
+
+## How to build an Android app with GraalVM Native Image 
+
+#### Requirements
+
+We will use Linux. On Windows, it is possible to follow this tutorial and get a working Android app [if you use WSL2](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux). 
+
+Download the latest GraalVM, Community Edition based on Java 11, from [here](https://github.com/graalvm/graalvm-ce-builds/releases). Set it up as your JVM in your `~/.bash_profile`:
+
+```
+export GRAALVM_HOME=<path to GraalVM home directory>
+export JAVA_HOME=$GRAALVM_HOME
+export PATH=$GRAALVM_HOME/bin:$PATH
+```
+
+When you type in `java -version` it should display something like this now:
+
+```
+> java -version
+openjdk version "11.0.10" 2021-01-19
+OpenJDK Runtime Environment GraalVM CE 21.0.0 (build 11.0.10+8-jvmci-21.0-b06)
+OpenJDK 64-Bit Server VM GraalVM CE 21.0.0 (build 11.0.10+8-jvmci-21.0-b06, mixed mode, sharing)
+```
+
+(The GraalVM version may differ)
+
+ Type `native-image` to check if it’s already there on the path. If not, install it with:
+
+```
+gu install native-image
+```
+
+`gu` should be available now in your console because of `$GRAALVM_HOME/bin` in your `PATH`. Also, [read this](https://www.graalvm.org/reference-manual/native-image/#prerequisites) and install whatever you need.
+
+You will need `adb` , „Android Debug Bridge”, to connect to your Android device and install the app on it. [Here you can find more on how to do it](https://www.fosslinux.com/25170/how-to-install-and-setup-adb-tools-on-linux.htm).
+
+Make sure your `gcc` is at least version 6. [You can try following these steps](https://tuxamito.com/wiki/index.php/Installing_newer_GCC_versions_in_Ubuntu). On top of that, you will need some specific C libraries (like GTK) to  build the native image and it varies from one computer to another, so I  can’t tell you exactly what to do. But it shouldn’t be a big problem.  Just follow error messages saying that you lack something and google how to install them. In my case this was the list:
```

Comment:
@julienrf : The errors can be very different depending on the platform and the environment. That's why I decided to just copy-paste the list of dependencies I had to install and add that comment that it can vary for everyone.